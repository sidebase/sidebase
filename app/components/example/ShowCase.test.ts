import { describe, expect, test } from 'vitest'
import ShowCase from './ShowCase.vue'
import { fireEvent, render } from '~/tests/testingLibraryVue'

describe('ShowCase', () => {
  test('mounts and is still the same', () => {
    const { html } = render(ShowCase, { props: { modelValue: '' } })
    expect(html()).toMatchSnapshot()
  })

  test('gets its starting input value from a prop', async () => {
    const modelValue = 'SIDESTREAM'
    const { getByPlaceholderText } = render(ShowCase, { props: { modelValue } })

    const nameInput = getByPlaceholderText('Your name...')
    expect(nameInput).toHaveValue(modelValue)
  })

  test('emits changed values on button click', async () => {
    const { emitted, getByPlaceholderText, getByText } = render(ShowCase, { props: { modelValue: '' } })

    // Enter text
    const name = 'Peter'
    const nameInput = getByPlaceholderText('Your name...')
    await fireEvent.update(nameInput, name)

    // Submit form
    const submitButton = getByText('Submit')
    await fireEvent.click(submitButton)

    // Check outcome
    const emits = emitted()
    expect(emits).toHaveProperty('submit')
    expect(emits.submit).toMatchObject([[name]])
  })
})
