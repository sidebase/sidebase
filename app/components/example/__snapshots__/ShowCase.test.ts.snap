// Vitest Snapshot v1

exports[`ShowCase > mounts and is still the same 1`] = `
"<div class=\\"p-1\\">
  <h1>Form:</h1>
  <p>An ant design form that emits a submit on click to the button. Note that we use is via an \\"auto-import\\", we did not need to perform a manual import in the scripts section.</p>
  <form>
    <div class=\\"flex\\"><span class=\\"ant-input-group-wrapper\\"><span class=\\"ant-input-wrapper ant-input-group\\"><span class=\\"ant-input-group-addon\\">Please enter your name:</span><input placeholder=\\"Your name...\\" type=\\"text\\" class=\\"ant-input\\">
      <!----></span></span><button class=\\"ant-btn ant-btn-primary\\" type=\\"submit\\">
        <transition-stub>
          <!---->
        </transition-stub><span>Submit</span>
      </button>
    </div>
  </form>
  <div class=\\"mt-2\\">
    <h1>Nuxt link:</h1>
    <p>This is a nuxt-link, a nuxt native component that we can easily use in histoire.</p>
    <nuxt-link-stub to=\\"/\\"></nuxt-link-stub>
  </div>
  <div class=\\"mt-2\\">
    <h1>Icon:</h1>
    <p>This is a svg-icon that was imported and turned into a pseudo-component, that we can style just like any other element.</p><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-24 fill-blue-200\\">
      <path d=\\"M18.966 6.239a.5.5 0 0 1 0 .707l-5.051 5.051 5.054 5.055a.5.5 0 0 1 0 .707l-1.21 1.21a.5.5 0 0 1-.707 0L12 13.913l-5.053 5.053a.5.5 0 0 1-.707 0L5.03 17.76a.5.5 0 0 1 0-.707L10.084 12 5.03 6.948a.5.5 0 0 1 0-.707l1.21-1.21a.5.5 0 0 1 .707 0L12 10.082l5.052-5.05a.5.5 0 0 1 .707 0l1.207 1.207z\\" fill-rule=\\"evenodd\\"></path>
    </svg>
  </div>
</div>"
`;
